library/entropy.c

```diff
@@ -576,10 +578,12 @@ int mbedtls_entropy_source_self_test( int verbose )
 int mbedtls_entropy_self_test( int verbose )
 {
     int ret = 1;
+#if !defined(MBEDTLS_TEST_NULL_ENTROPY)
```

Comment:
I reviewed the changes and tested it, +1 from me either way. (I like @andresag01 's idea and I am happy to apply it before merge if you want me to)